src/Chapter3.hs

```diff
@@ -374,6 +379,32 @@ after the fight. The battle has the following possible outcomes:
 
 -}
 
+data Knight = Knight
+  { knightName      :: String
+  , knightHealth :: Int
+  , knightAttack :: Int
+  , knightGold :: Int
+  } deriving (Show)
+
+data Monster = Monster
+  { monsterName      :: String
+  , monsterHealth :: Int
+  , monsterAttack :: Int
+  , monsterGold :: Int
+  } deriving (Show)
+
+fight :: Knight -> Monster -> Int
+fight k m
+  | knightAttack k <= 0 && monsterAttack m <= 0 = knightGold k
```

Comment:
Very nice catch on negatives here. 
But I think in this case it would be better to use `Natural` for all of the fields, as gold also could be negative, which is not good 🙂